Force an agent return if there is an error on reload (#25721)
hashicorp/nomad: website/content/docs/upgrade/upgrade-specific.mdx

peer-remove`](/nomad/docs/commands/operator/raft/remove-peer) command, and the
`address` parameter for the `DELETE /v1/operator/raft/peer` API.

> Errors encountered when reloading agent configuration will now cause agents to
> exit. Before configuration errors during reloads were only logged. This could 
> lead to agents running but unable to communicate. Any other errors when parsing
> the new configuration will be logged and the reload will be aborted, consistent
> with the current behavior.

Reviewer: ```suggestion
Errors encountered when reloading agent configuration now cause agents to exit.
In prior versions, Nomad only logged configuration errors during reloads. This
could lead to agents running but unable to communicate. Any other errors when
parsing the new configuration are logged and the reload is aborted, consistent
with the current behavior.
```

changed to use present tense
